library/src/ui/table/editMode.js

```diff
@@ -29,22 +29,40 @@ window.CoNDeT.ui.TableComponentEditMode = (function () {
   constructor.prototype.onMouseLeave = function (event) {
     this.updateXY(event);
     this.resetState();
-    this.table.ref.className = "condet-table condet-class-" + this.table.props.class;
+    this.table.ref.className =
+      "condet-table condet-class-" + this.table.props.class;
   };
   constructor.prototype.onMouseEnter = function () {
-    this.table.ref.className = "condet-table condet-class-" + this.table.props.class + " condet-table-movable";
-  }
+    this.table.ref.className =
+      "condet-table condet-class-" +
+      this.table.props.class +
+      " condet-table-movable";
+  };
 
   constructor.prototype.onInit = function (table) {
     this.table = table;
+    this.table.ref.appendChild(this.removeTableElement(table.id));
```

Comment:
Nie zadziała ze zmianą trybu, bo wstawiasz to jednorazowo. Zrób komponent np. DeleteTableBtnComponent i przenieś tam logikę odpowiedzialną za tego spana. I w komponencie `NameComponent` renderuj go tylko w trybie edycji. Aby móc usunąć tabelę, podaj id tabeli w dół jako propsy `NameComponent`. Możesz przenieść logikę eventów do oddzielnej strategii (ale ponieważ on i tak będzie renderowany w trybie edycji to nie musisz)